facebook/delphiFacebook/R/contingency_variables.R

```diff
@@ -951,3 +1059,51 @@ code_behaviors <- function(input_data, wave) {
   
   return(input_data)
 }
+
+#' Activities
+#'
+#' @param input_data input data frame of raw survey data
+#' @param wave integer indicating survey version
+#' 
+#' @return augmented data frame
+code_addl_activities <- function(input_data, wave) {
+  if ("C13a" %in% names(input_data)) {
+    # introduced in wave 4
+    activities <- split_options(input_data$C13a)
+    
+    input_data$a_mask_work_outside_home_1d <- is_selected(activities, "1")
+    input_data$a_mask_shop_1d <- is_selected(activities, "2")
+    input_data$a_mask_restaurant_1d <- is_selected(activities, "3")
+    input_data$a_mask_spent_time_1d <- is_selected(activities, "4")
+    input_data$a_mask_large_event_1d <- is_selected(activities, "5")
+    input_data$a_mask_public_transit_1d <- is_selected(activities, "6")
+  } else {
+    input_data$a_mask_work_outside_home_1d <- NA
+    input_data$a_mask_shop_1d <- NA
+    input_data$a_mask_restaurant_1d <- NA
+    input_data$a_mask_spent_time_1d <- NA
+    input_data$a_mask_large_event_1d <- NA
+    input_data$a_mask_public_transit_1d <- NA
+  }
+  
+  if ("C13c" %in% names(input_data)) {
+    # introduced in wave 10 as "indoors" activities version of C13a
+    activities <- split_options(input_data$C13c)
+    
+    input_data$a_mask_work_outside_home_indoors_1d <- is_selected(activities, "1")
+    input_data$a_mask_shop_indoors_1d <- is_selected(activities, "2")
+    input_data$a_mask_restaurant_indoors_1d <- is_selected(activities, "3")
+    input_data$a_mask_spent_time_indoors_1d <- is_selected(activities, "4")
+    input_data$a_mask_large_event_indoors_1d <- is_selected(activities, "5")
+    input_data$a_mask_public_transit_1d <- is_selected(activities, "6")
+  } else {
+    input_data$a_mask_work_outside_home_indoors_1d <- NA
+    input_data$a_mask_shop_indoors_1d <- NA
+    input_data$a_mask_restaurant_indoors_1d <- NA
+    input_data$a_mask_spent_time_indoors_1d <- NA
+    input_data$a_mask_large_event_indoors_1d <- NA
+  }
+  
+  return(input_data)
+}
```

Comment:
I don't think these indicators measure something useful. Consider C13b and C13c on Wave 11. Let's say a respondent says, in C13b, that they went to an indoor market, grocery store, or pharmacy (option 2) in the past 24 hours. C13c will then ask if they wore a mask *during only that activity*, i.e. option 2 will be the only listed option. But they will count as a `FALSE` for whether they wore a mask on public transit or attending an indoor event, despite doing neither.

If we want "what % of people conducting this activity are wearing masks?", which I assume is what Facebook is thinking of, then you'll need to account for what they selected in C13/C13b when evaluating C13a/C13c.

Reply:
Thanks, good catch.